main/src/library/flix/actor/ActorSystem.flix

```diff
@@ -0,0 +1,183 @@
+/// @field `startPolicy`     that describes the policy for how actors are started.
+/// @field `restartPolicy`   that describes the policy for how actors are restarted if they crash.
+/// @field `shutdownPolicy`  that describes the policy for how actors are stopped.
+/// @field `control`         the runtime control channel of this actor system
+/// @field `actors`          the actors that are running
+///
+type alias ActorSystem = {
+    actors: ActorPolicy,
+    startPolicy: ActorPolicy,
+    restartPolicy: ActorPolicy,
+    shutdownPolicy: ActorPolicy
+}
+
+///
+/// A running actor system is a type alias for a channel of actor events.
+///
+type alias RunningActorSystem = Channel[SystemEvent]
+
+///
+/// An actor is identified by its mailbox
+///
+type alias ActorRef = Mailbox // TODO: Quentin: Maybe a name and a mailbox?
+
+namespace ActorSystem {
+
+    ///
+    /// Returns an actor system with a default configuration.
+    ///
+    pub def default(): ActorSystem = {
+        actors          = #{},
+        startPolicy     = defaultStartPolicy(),
+        shutdownPolicy  = defaultShutdownPolicy(),
+        restartPolicy   = defaultRestartPolicy()
+    }
+
+    ///
+    /// Returns the default mailbox capacity.
+    ///
+    pub def defaultMailboxCapacity(): Int = 1000 * 1000
+
+    enum InternalActorState {
+      ///
+      /// An actor that has not been started previously, and has no mailbox
+      ///
+      case NotStarted,
+      ///
+      /// An actor that has been started and been assigned a mailbox. The actor could be stopped or crashed now.
+      ///
+      case Started(State, ActorRef)
+    }
+
+    def statesOfActors(actors: Map[Str, (InternalActorState, Actor)]): ActorPolicy =
+      Map.foldLeftWithKey((actorName, states, actor) ->
+        match fst(actor) {
+          case NotStarted => states <+> #{ActorState(actorName, Stopped).}
+          case Started(st, _) => states <+> #{ActorState(actorName, st).}
+        }, #{}, actors)
+
+
+  def actorSystem(s: ActorSystem, actors: Map[Str, (InternalActorState, Actor)], control: Channel[SystemEvent]): Unit = {
+      match <- control {
+        case SystemStart =>
+          let currentState = statesOfActors(actors);
+          let actors2 = fold Start actors ((actorName, m) -> match Map.get(actorName, actors) {
+            case None =>
+              // TODO: inconsistent policy, starts a non-existing actor
+              actors
+            case Some(NotStarted, behavior) =>
+              // Start the actor with a new mailbox
+              let c = chan SystemEvent 10;
+              let actorRef = startActor(c, actorName, behavior);
+              let readyChan = chan Ready 0;
+              actorRef <- Start(readyChan);
+              <- readyChan; // TODO: we block here, but what if the actor never sends Ready? -> select with default? Have an internal actor state that is "started but hasn't replied Ready yet"?
```

Comment:
What about spawning a new process that will send the message on the control channel when it is received?
That we we dont block the main loop.